firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/LiveGenerativeModel.kt

```diff
@@ -0,0 +1,122 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.vertexai
+
+import com.google.firebase.FirebaseApp
+import com.google.firebase.appcheck.interop.InteropAppCheckTokenProvider
+import com.google.firebase.auth.internal.InternalAuthProvider
+import com.google.firebase.vertexai.common.APIController
+import com.google.firebase.vertexai.common.AppCheckHeaderProvider
+import com.google.firebase.vertexai.type.BidiGenerateContentClientMessage
+import com.google.firebase.vertexai.type.BidiServerHandshakeFailed
+import com.google.firebase.vertexai.type.Content
+import com.google.firebase.vertexai.type.LiveGenerationConfig
+import com.google.firebase.vertexai.type.LiveSession
+import com.google.firebase.vertexai.type.RequestOptions
+import com.google.firebase.vertexai.type.Tool
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.okhttp.OkHttp
+import io.ktor.client.plugins.websocket.WebSockets
+import io.ktor.client.plugins.websocket.webSocketSession
+import io.ktor.websocket.Frame
+import io.ktor.websocket.close
+import io.ktor.websocket.readBytes
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * Represents a multimodal model (like Gemini) capable of real-time content generation based on
+ * various input types, supporting bidirectional streaming.
+ */
+public class LiveGenerativeModel
+internal constructor(
+  private val modelName: String,
+  private val config: LiveGenerationConfig? = null,
+  private val tools: List<Tool>? = null,
+  private val systemInstruction: Content? = null,
+  private val location: String,
+  private val controller: APIController
+) {
+  internal constructor(
+    modelName: String,
+    apiKey: String,
+    firebaseApp: FirebaseApp,
+    config: LiveGenerationConfig? = null,
+    tools: List<Tool>? = null,
+    systemInstruction: Content? = null,
+    location: String = "us-central1",
+    requestOptions: RequestOptions = RequestOptions(),
+    appCheckTokenProvider: InteropAppCheckTokenProvider? = null,
+    internalAuthProvider: InternalAuthProvider? = null,
+  ) : this(
+    modelName,
+    config,
+    tools,
+    systemInstruction,
+    location,
+    APIController(
+      apiKey,
+      modelName,
+      requestOptions,
+      "gl-kotlin/${KotlinVersion.CURRENT} fire/${BuildConfig.VERSION_NAME}",
+      firebaseApp,
+      AppCheckHeaderProvider(TAG, appCheckTokenProvider, internalAuthProvider),
+    ),
+  )
+
+  /**
+   * Returns a LiveSession object using which you could send/receive messages from the server
+   * @return LiveSession object created. Returns null if the object cannot be created.
+   * @throws [BidiServerHandshakeFailed] if the handshake with the server failed.
+   */
+  public suspend fun connect(): LiveSession {
+    val client = HttpClient(OkHttp) { install(WebSockets) }
```

Comment:
This should be here?

https://github.com/firebase/firebase-android-sdk/blob/8a72ed5841a203e48b6c1d771f52cc71110834cd/firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/common/APIController.kt#L124

You don't need a new client every time